applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_truss_element_3D2N.h

```diff
@@ -33,24 +39,51 @@ class AdjointFiniteDifferenceTrussElement : public AdjointFiniteDifferenceTrussE
 
     void Calculate(const Variable<Vector >& rVariable,
                         Vector& rOutput,
-                        const ProcessInfo& rCurrentProcessInfo) override;                    
+                        const ProcessInfo& rCurrentProcessInfo) override;
 
+    /**
+     * Calculates the derivative of stresses/stress resultants w.r.t primal displacement. The calculation is done analytically.
+     * The derivative consists of two parts: The analytic derivative of the current length w.r.t. displacement
+     * and an individual pre-factor for the different stresses/stress resultants.
+     */
     void CalculateStressDisplacementDerivative(const Variable<Vector>& rStressVariable,
                                     Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo) override;
 
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
 private:
+    void CheckVariables();
+
+    void CheckDofs();
+
+    void CheckProperties(const ProcessInfo& rCurrentProcessInfo);
+
     double GetPerturbationSizeModificationFactor(const Variable<array_1d<double,3>>& rDesignVariable) override;
 
     double CalculateReferenceLength();
 
     double CalculateCurrentLength();
 
+    /**
+     * Calculates the derivative of the current length w.r.t. primal displacements. This derivative is part of the
+     * analytic stress displacement derivative.
```

Comment:
IMO the stresses do not need to be mentioned here. You could use this in any other occasion too.